components/camel-aws/camel-aws2-ddb/src/main/java/org/apache/camel/component/aws2/ddbstream/Ddb2StreamConsumer.java

```diff
@@ -33,14 +35,13 @@
 import software.amazon.awssdk.services.dynamodb.model.GetRecordsRequest;
 import software.amazon.awssdk.services.dynamodb.model.GetRecordsResponse;
 import software.amazon.awssdk.services.dynamodb.model.Record;
-import software.amazon.awssdk.services.dynamodb.streams.DynamoDbStreamsClient;
 
 public class Ddb2StreamConsumer extends ScheduledBatchPollingConsumer {
 
     private static final Logger LOG = LoggerFactory.getLogger(Ddb2StreamConsumer.class);
 
     private final ShardIteratorHandler shardIteratorHandler;
-    private String lastSeenSequenceNumber;
+    private final Map<String, String> lastSeenSequenceNumbers = new HashMap<>();
```

Comment:
Is there any code that removes entries from this map, in case it grows forever big? Or would there only be a little number of entries?

Reply:
I'd expect the number of entries to always remain quite low. The highest number of shards I've personally seen in practice is 7.